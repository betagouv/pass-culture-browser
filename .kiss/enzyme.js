import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
