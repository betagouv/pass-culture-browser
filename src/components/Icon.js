// look here https://gorangajic.github.io/react-icons/md.html
// for icons
import * as reactIconPack from 'react-icons/lib/md';

export default ( props ) => {
  const iconName = 'Md' + props.name.replace(/(^|-)(\w)/g, (m0, m1, m2) =>
    m2.toUpperCase());
  return reactIconPack[iconName]();
}
