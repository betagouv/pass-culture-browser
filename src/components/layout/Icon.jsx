import PropTypes from 'prop-types'
import React from 'react'

 import { ICONS_URL } from '../../utils/config'

 const Icon = ({ alt, src, svg, ...imgProps }) => (
   <img
     {...imgProps}
     alt={alt || svg}
     src={src || `${ICONS_URL}/${svg}.svg`}
   />
)

 Icon.defaultProps = {
  className: null,
  src: null,
  svg: null
}

 Icon.propTypes = {
  className: PropTypes.string,
  src: PropTypes.string,
  svg: PropTypes.string
}

 export default Icon
