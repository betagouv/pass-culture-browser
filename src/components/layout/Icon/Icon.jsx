import PropTypes from 'prop-types'
import React from 'react'

import { ICONS_URL } from '../../../utils/config'

const Icon = ({ alt, src, svg, png, ...imgProps }) => (
  <img
    {...imgProps}
    alt={alt || svg}
    src={src || `${ICONS_URL}/${svg ? `${svg}.svg` : `${png}.png`}`}
  />
)

Icon.defaultProps = {
  alt: null,
  className: null,
  png: null,
  src: null,
  svg: null,
}

Icon.propTypes = {
  alt: PropTypes.string,
  className: PropTypes.string,
  png: PropTypes.string,
  src: PropTypes.string,
  svg: PropTypes.string,
}

export default Icon
