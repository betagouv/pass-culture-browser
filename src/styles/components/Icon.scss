.icon {
  width: 25px;
  height: 25px;
  fill: inherit;
  vertical-align: middle;
  margin-left: auto;
  margin-right: auto;
}
