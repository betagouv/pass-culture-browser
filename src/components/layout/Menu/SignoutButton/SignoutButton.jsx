import PropTypes from 'prop-types'
import React from 'react'

import Icon from '../../Icon'

const SignoutButton = ({ onSignoutClick, ...props }) => (
  <button
    className="pc-text-button flex-columns text-left px16 py8"
    id="main-menu-logout-button"
    onClick={onSignoutClick(props)}
    type="button"
  >
    <span className="menu-icon mr16 text-center">
      <Icon
        alt=""
        svg="ico-deconnect-w"
      />
    </span>
    <span className="pt5">{'DÃ©connexion'}</span>
  </button>
)

SignoutButton.propTypes = {
  onSignoutClick: PropTypes.func.isRequired,
}

export default SignoutButton
