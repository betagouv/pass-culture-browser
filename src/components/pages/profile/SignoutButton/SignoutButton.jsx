import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'

import Icon from '../../../layout/Icon/Icon'

const noOp = () => false

const SignoutButton = ({ historyPush, onSignOutClick, readRecommendations }) => (
  <Link
    onClick={onSignOutClick(historyPush, readRecommendations)}
    to={noOp}
  >
    <Icon svg="ico-signout" />
    <div className="list-link-label">
      {'DÃ©connexion'}
    </div>
    <Icon svg="ico-next-lightgrey" />
  </Link>
)

SignoutButton.propTypes = {
  historyPush: PropTypes.func.isRequired,
  onSignOutClick: PropTypes.func.isRequired,
  readRecommendations: PropTypes.PropTypes.arrayOf(PropTypes.shape()).isRequired,
}

export default SignoutButton
