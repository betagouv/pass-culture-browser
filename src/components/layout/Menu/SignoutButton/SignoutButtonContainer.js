import { connect } from 'react-redux'
import { requestData, reinitializeData } from 'redux-thunk-data'

import SignoutButton from './SignoutButton'
import { toggleMainMenu } from '../../../../reducers/menu'

export const mapDispatchToProps = dispatch => ({
  onSignoutClick: ({ history, readRecommendations }) => () => {
    const handleRequestSignout = () => {
      const handleSuccessAfterSignout = () => {
        history.push('/connexion')
        dispatch(toggleMainMenu())
        dispatch(reinitializeData({ excludes: ['features'] }))
      }
      dispatch(
        requestData({
          apiPath: '/users/signout',
          handleSuccess: handleSuccessAfterSignout,
        })
      )
    }

    if (!readRecommendations || readRecommendations.length === 0) {
      handleRequestSignout()
    } else {
      dispatch(
        requestData({
          apiPath: '/recommendations/read',
          body: readRecommendations,
          handleFail: handleRequestSignout,
          handleSuccess: handleRequestSignout,
          method: 'PUT',
        })
      )
    }
  },
})

export default connect(
  null,
  mapDispatchToProps
)(SignoutButton)
