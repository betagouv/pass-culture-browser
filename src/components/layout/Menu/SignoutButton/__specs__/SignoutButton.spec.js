import { shallow } from 'enzyme'
import React from 'react'

import SignoutButton from '../SignoutButton'
import Icon from '../../../Icon'

describe('src | components | menu | SignoutButton', () => {
  let props

  beforeEach(() => {
    props = {
      onSignoutClick: jest.fn(),
    }
  })

  it('should match the snapshot', () => {
    // given
    const wrapper = shallow(<SignoutButton {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  describe('render()', () => {
    it('should render one button and one Icon', () => {
      // given
      const wrapper = shallow(<SignoutButton {...props} />)

      // when
      const button = wrapper.find('button')
      const icon = wrapper.find(Icon)
      const text = wrapper.find('.pt5').text()

      // then
      expect(button).toHaveLength(1)
      expect(icon).toHaveLength(1)
      expect(text).toBe('DÃ©connexion')
    })
  })
})
