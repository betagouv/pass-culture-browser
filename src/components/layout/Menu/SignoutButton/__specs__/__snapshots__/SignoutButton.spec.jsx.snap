// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | menu | SignoutButton should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignoutButton
    onSignoutClick={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="menu-icon mr16 text-center"
        >
          <Icon
            alt=""
            className={null}
            src={null}
            svg="ico-deconnect-w"
          />
        </span>,
        <span
          className="pt5"
        >
          Déconnexion
        </span>,
      ],
      "className": "pc-text-button flex-columns text-left px16 py8 fs16",
      "id": "main-menu-logout-button",
      "onClick": undefined,
      "type": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Icon
            alt=""
            className={null}
            src={null}
            svg="ico-deconnect-w"
          />,
          "className": "menu-icon mr16 text-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "",
            "className": null,
            "src": null,
            "svg": "ico-deconnect-w",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Déconnexion",
          "className": "pt5",
        },
        "ref": null,
        "rendered": "Déconnexion",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="menu-icon mr16 text-center"
          >
            <Icon
              alt=""
              className={null}
              src={null}
              svg="ico-deconnect-w"
            />
          </span>,
          <span
            className="pt5"
          >
            Déconnexion
          </span>,
        ],
        "className": "pc-text-button flex-columns text-left px16 py8 fs16",
        "id": "main-menu-logout-button",
        "onClick": undefined,
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              alt=""
              className={null}
              src={null}
              svg="ico-deconnect-w"
            />,
            "className": "menu-icon mr16 text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "",
              "className": null,
              "src": null,
              "svg": "ico-deconnect-w",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Déconnexion",
            "className": "pt5",
          },
          "ref": null,
          "rendered": "Déconnexion",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
